internal/ir/builder.go

```diff
@@ -252,15 +252,69 @@ func (b *builder) assignStmt(fn *Function, lhss, rhss []ast.Expr, syntax *ast.As
 func (b *builder) assign(fn *Function, lhs, rhs ast.Expr, syntax *ast.AssignStmt) {
 	switch lhs := lhs.(type) {
 	case *ast.Ident:
-		if closure, isFuncLit := rhs.(*ast.FuncLit); isFuncLit {
-			fn.emit(b.funcLit(fn, lhs.Name, closure))
+		b.createValueByRHSType(fn, lhs, rhs, syntax)
+	default:
+		unsupportedNode(lhs)
+	}
+}
+
+// createValueByRHSType according to the type of the right operand, a different function will be executed to create
+// the var.
+func (b *builder) createValueByRHSType(fn *Function, lhs *ast.Ident, rhs ast.Expr, syntax *ast.AssignStmt) {
```

Comment:
I think that we can improve this function name and docs:

```go
// assingValue create a new Value according to the Value yielded from rhs expression.
//
// If the rhs expression is a *ast.FuncLit a new Closure Instruction will be created and emited
// to fn. If the rhs expression is a identifier, the identifier value will be resolve to set 
// the identifier value as the value of lhs identifier. Otherwise a new named local variable
// will be created using the Value created from rhs expression.
func (b *builder) assignValue(fn *Function, lhs *ast.Ident, rhs ast.Expr, syntax *ast.AssignStmt)
```